astropy/coordinates/baseframe.py

```diff
@@ -454,10 +454,63 @@ def representation_component_units(self):
                 out[repr_name] = repr_unit
         return out
 
+    def replicate(self, copy=False, **kwargs):
+        """
+        Generates a new frame object that is a copy of this one, but possibly
+        with frame attributes overriden and/or by reference.
+
+        If ``copy`` is set to `True` then a full copy of the internal arrays
+        will be made.  By default the replica will use a reference to the
+        original arrays when possible to save memory.  The internal arrays
+        are normally not changeable by the user so in most cases it should not
+        be necessary to set ``copy`` to `True`.
+
+        Parameters
+        ----------
+        copy : bool, optional
+            If True, the resulting object is a copy of the data.  When False,
+            references are used where  possible. This rule also applies to the
+            frame attributes.
+
+        Any additional keywords are treated as frame attributes to be set on the
+        new frame object.
+
+        Returns
+        -------
+        frameobj : same as this frame
+            Replica of this object, but possibly with new frame attributes.
+        """
+        return self._apply('copy' if copy else 'replicate', **kwargs)
+
+    def replicate_without_data(self, copy=False, **kwargs):
+        """
+        Generates a new frame object that is a copy of this one, but without
```

Comment:
Similarly, maybe
```
Return a replicate without data, optionally overriding some attributes.
```
(and then still some text like you have here, including the analogy with `realize_frame`)